src/Zforce.cpp

```diff
@@ -718,6 +760,74 @@ void Zforce::ParseTouchDescriptor(TouchDescriptorMessage* msg, uint8_t* payload)
   }
 }
 
+void Zforce::ParsePlatformInformation(PlatformInformationMessage *msg, uint8_t *rawData, uint32_t length)
+{
+  uint16_t value = 0;
+  uint16_t valueLength = 0;
+  const uint8_t offset = 10;
+
```

Comment:
This seems to jump ahead 10 bytes from the beginning of the whole payload (including I2C header). I don't seem to find a place where it takes into account that the length is longer than 127, neither in the first ASN.1 length, DeviceInformation Length, or PlatformInformation length. The first, first+second, or all 3 could be longer than 127.

Reply:
Based on the response from the PlatformInformation request I actually assumed that the lenghts for all three of them would be longer than 127 but less than 256 but assumptions are rarely the good choice to be made. 
I have updated the implementation now to take the varying lengths into account. Hopefully a more maintainable solution.